plumbing/object/commit_walker.go

```diff
@@ -197,26 +197,38 @@ func NewCommitAllIter(repoStorer storage.Storer, commitIterFunc func(*Commit) Co
 	commitsPath := list.New()
 	commitsLookup := make(map[plumbing.Hash]*list.Element)
 	head, err := storer.ResolveReference(repoStorer, plumbing.HEAD)
-	if err != nil {
+	if err != nil && err != plumbing.ErrReferenceNotFound {
 		return nil, err
+	} else if err != plumbing.ErrReferenceNotFound {
```

Comment:
Does it make sense?

Reply:
maybe it should be `err == plumbing.ErrReferenceNotFound`

Reply:
@kuba-- I only want to add the head reference if I actually have the reference, so only if the error is not plumbing.ErrReferenceNotFound

Maybe I can check if `err == nil` or `head != nil`

Reply:
So, maybe check if head is not nil (is it possible)?
Because if err is not plumbing.ErrReferenceNotFound does not mean that err != nil.
My question is - shall we consider just plumbing.ErrReferenceNotFound? What do we want to do with other errors?

Maybe what we had at the beginning `err != nil` is good enough?